### PR TITLE
Add a quick explanation of compression formats

### PR DESCRIPTION
**Pull request by alexwlchan**
Rendered version: https://github.com/wellcomecollection/storage-service/blob/explain-compression/docs/explanations/compression-formats.md

### CODE REVIEW

File: docs/explanations/compression-formats.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+-   Because compressed bags can be passed around atomically.
+    A user has either finished uploading a bag, or they haven't.
+    You can't have a partial bag upload.
+
+Why do we store bags as uncompressed files?
+
+-   Our digitised bags are mostly made up of a small number of medium-sized images.
+    (Typically a few hundred files per bag, each 5–10MB.)
+    It's useful to be able to address these images individually, both programatically and for humans browsing our S3 buckets by hand.
```
Comment (jamesgorrie) [inline docs/explanations/compression-formats.md:15]: "By hand"? Isn't it the human aspect that's important, not which limb?

File: docs/explanations/compression-formats.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Each entry is prepended by a header with some information about the name and size of that entry.
+
+![](tar_gz_format.png)
+
+This is much easier to read from S3 -- you start reading bytes and keep going until you've read all of them.
+We don't need to provision any EBS storage, and we can read an arbitrarily large bag with this approach.
+
+It was simpler to update our workflow tools to send tar.gz archives than get zip unpacking working, so that's what we did.
+
+Uploading zip files as the initial bag is currently unsupported.
```
Comment (jamesgorrie) [inline docs/explanations/compression-formats.md:56]: I think I'm clear, but you mean we won't compress a zip into a tar.gz?

**Review by jamesgorrie — APPROVED** [review]
Hard to tel lif you're missing anything as I am unfamiliar with the service, but it is very clear and I understand what's here.

**Review by kenoir — APPROVED** [review]

File: docs/explanations/compression-formats.md (hunk 2 above)
Comment (alexwlchan) [reply docs/explanations/compression-formats.md]: Clearer now?

**Pull request merged**